app/src/reports/VariableDisparityReport.tsx

```diff
@@ -17,91 +10,133 @@ import ToggleButton from "@material-ui/lab/ToggleButton";
 import ToggleButtonGroup from "@material-ui/lab/ToggleButtonGroup";
 import Alert from "@material-ui/lab/Alert";
 import { Fips } from "../utils/madlib/Fips";
+import {
+  METRIC_CONFIG,
+  VariableConfig,
+  MetricConfig,
+} from "../data/MetricConfig";
 
-// TODO - remove hardcoded values when we have full support
-const SUPPORTED_MADLIB_VARIABLES: DropdownVarId[] = ["covid"];
-
-function DisVarGeo(props: {
+function VariableDisparityReport(props: {
   dropdownVarId: DropdownVarId;
   fips: Fips;
   updateFipsCallback: Function;
   vertical?: boolean;
 }) {
   // TODO Remove hard coded fail safe value
-  const [metric, setMetric] = useState<MetricToggle>(
-    SUPPORTED_MADLIB_VARIABLES.includes(props.dropdownVarId)
-      ? (METRICS_FOR_VARIABLE[props.dropdownVarId as string][0] as MetricToggle)
-      : ("covid_cases" as MetricToggle)
+  const [variableConfig, setVariableConfig] = useState<VariableConfig | null>(
+    Object.keys(METRIC_CONFIG).includes(props.dropdownVarId)
+      ? METRIC_CONFIG[props.dropdownVarId as string][0]
+      : null
   );
 
+  useEffect(() => {
+    setVariableConfig(
+      Object.keys(METRIC_CONFIG).includes(props.dropdownVarId)
+        ? METRIC_CONFIG[props.dropdownVarId as string][0]
+        : null
+    );
+  }, [props.dropdownVarId]);
+
+  const fields: VariableId[] = [];
+  if (variableConfig && variableConfig.metrics["per100k"]) {
+    fields.push(variableConfig.metrics["per100k"].metricId as VariableId);
+  }
+  if (variableConfig && variableConfig.metrics["pct_share"]) {
+    fields.push(variableConfig.metrics["pct_share"].metricId as VariableId);
+  }
+
+  const tableFields = variableConfig
+    ? [...fields, "population" as VariableId, "population_pct" as VariableId]
```

Comment:
Are these typecasts necessary if you just type the tableFields variable instead? It should recognize the types then